transaction/core/src/signed_contingent_input.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+//! A signed contingent input as described in MCIP #31
+
+use crate::{
+    ring_signature::{
+        CurveScalar, Error as RingSignatureError, GeneratorCache, KeyImage, OutputSecret,
+        PresignedInputRing, RingMLSAG, SignedInputRing,
+    },
+    tx::TxIn,
+    Amount, Commitment, CompressedCommitment, TokenId,
+};
+use alloc::vec::Vec;
+use displaydoc::Display;
+use prost::Message;
+
+/// The "unmasked" data of an amount commitment
+#[derive(Clone, Eq, Message, PartialEq)]
+pub struct UnmaskedAmount {
+    /// The value of the amount commitment
+    #[prost(fixed64, tag = "1")]
+    pub value: u64,
+
+    /// The token id of the amount commitment
+    #[prost(fixed64, tag = "2")]
+    pub token_id: u64,
+
+    /// The blinding factor of the amount commitment
+    #[prost(message, required, tag = "3")]
+    pub blinding: CurveScalar,
```

Comment:
It's used in line 101. When the code goes through and validates the blinding aligns with the commitments in input_rules. I am not sure why we need a blinding factor though, since if we're publishing these unmasking fields, wouldn't it be equivalently unblinded?

Reply:
Don't we publish the unmasking amount fields though? If the unmasking amount contains the blinding factor, don't we have to assume that the information is public?

Reply:
I think Chris is saying since the blinding factor is used only to blind the amount, and we're revealing the amount, there's no harm in revealing that blinding factor.

Reply:
Right, my point is that if we are revealing the blinding factor, we could just have a convention for these that the blinding factor is 0 and exclude it from calculations.

Reply:
I'm not all the way through the mlsag & bulletproof math, but when you go through those, making the curve scalar zero would make any arbitrary counterparty amount work as valid when it's run through the proofs.

Reply:
I think this question is answered adequately here: https://github.com/mobilecoinfoundation/mobilecoin/pull/1871#discussion_r867258634